sdk/core/allocator/token_unseal.S

```diff
@@ -0,0 +1,73 @@
+#include <cheri-builtins.h>
+#include "token.h"
+
+/*
+ * void *
+ * token_obj_unseal_internal(struct SKeyStruct *ca0, struct SObjStruct *ca1)
+ *
+ * Register allocation:
+ *
+ *  - ca0 holds a sealing key, either the user's or the real deal, and is
+ *    replaced with the unsealed value or NULL
+ *
+ *  - ca1 holds the user's sealed object pointer
+ *
+ *  - t0/ct0 holds a copy of the user key
+ *
+ *  - t1/ct1 is used within each local computation and never holds secrets
+ */
+
+.global token_obj_unseal_internal
+token_obj_unseal_internal:
+
+  // Verify key tag
+  cgettag t1, ca0
+  beqz    t1, 9f
+
+  // Verify key address == base and len > 0
+  cgetbase t1, ca0 
+  bne      a0, t1, 9f // as-integer access to ca0 gives address
```

Comment:
Why are we checking that the base == address?  We probably could optimise for size later by providing compartments with ranges of sealing keys.

The check that we might want is that the address > 16, to make sure that it's a software key and we don't accidentally allow unsealing zeroed things.

Reply:
I copied the test from https://github.com/microsoft/cheriot-rtos/blob/b9586f31616fb91b8e9060cd5bf84428300e90a8/sdk/core/allocator/token.h#L38-L44 .  I'm happy to change it, but that one might also want to be changed.

Reply:
Oh, right.  The actual check that we want is that address is in bounds, but base == address is simpler and works for everything we do for now.